astropy/io/fits/tests/test_convenience.py

```diff
@@ -61,3 +60,36 @@ def test_table_to_hdu(self, tmpdir):
         assert isinstance(hdu, fits.BinTableHDU)
         filename = str(tmpdir.join('test_table_to_hdu.fits'))
         hdu.writeto(filename, overwrite=True)
+
+    def test_printdiff(self):
+        """
+        Test that FITSDiff can run the different inputs without crashing.
+        """
+
+        # Testing different string input options
+        assert printdiff(self.data('arange.fits'), self.data('blank.fits')) is None
```

Comment:
Per our in person discussion the other week, I feel like it would be wrong to test the actually printed output.  I do not want my tests dependent on how FITSDiff decides to format it's report string.  Or is this not what you mean...